ko/sql/jsp.rst

```diff
@@ -504,90 +725,431 @@ CUBRID에서는 **java.sql.ResultSet** 을 반환하는 Java 저장 함수/프
         }
     }
 
-.. note::
+.. _jsp-get-client-info:
 
-    **ResultSet** 은 입력 인자로 사용할 수 없으며, 이를 IN 인자로 전달할 경우에는 에러가 발생한다. Java가 아닌 환경에서 **ResultSet** 을 반환하는 함수를 호출할 경우에도 에러가 발생한다.
+연결 중인 클라이언트 정보 획득
+--------------------------------
 
-Java 저장 함수/프로시저에서 Set 타입의 IN/OUT
----------------------------------------------
+.. code-block:: sql
 
-CUBRID의 Java 저장 함수/프로시저에서 Set 타입이 IN OUT인 경우 Java에서 인자 값을 변경할 경우 변경 값이 전달이 되도록 Set 타입이 OUT 인자로 전달될 때는 2차원 배열로 전달하도록 해야 한다.
+    CREATE OR REPLACE FUNCTION sp_client_info () RETURN STRING as language java name 'SpTestClientInfo.getClientInfo() return java.lang.String'; 
+
+.. code-block:: java
+
+    import java.util.Properties;
+    import java.sql.*;
+     
+    public class SpTestClientInfo {
+        public static String getClientInfo() {
+            Connection conn = null;
+            String result = "";
+     
+            try {
+                conn = DriverManager.getConnection("jdbc:default:connection:");
+     
+                Properties props = conn.getClientInfo();
+
+                // How to get from the Properties
+                // String user = props.getProperty ("user");
+
+                result = props.toString ();
+            } catch (Exception e) {
+                result = e.getMessage ();
+            }
+            return result;
+        }
+    }
 
 .. code-block:: sql
 
-    CREATE PROCEDURE setoid(x in out set, z object) AS LANGUAGE JAVA 
-    NAME 'SetOIDTest.SetOID(cubrid.sql.CUBRIDOID[][], cubrid.sql.CUBRIDOID';
+    SELECT sp_client_info ();
+
+    sp_client_info()
+    ======================
+    '{pid=200270, user=DBA, login=cubrid, program=csql, type=2, host=cubrid, ip=192.168.2.201}'
+
+다른 데이터베이스 연결
+======================
+
+서버 측 JDBC 드라이버를 사용하더라도 현재 연결된 데이터베이스를 사용하지 않고, 외부의 다른 데이터베이스에 연결할 수도 있다. 
+외부의 데이터베이스에 대한 Connection을 얻는 것은 일반적인 JDBC Connection과 다르지 않다. 이에 대한 자세한 내용은 JDBC API를 참조한다.
+
+.. warning::
+
+    다른 데이터베이스에 연결하는 경우, Java 메서드의 수행이 종료되더라도 CUBRID 데이터베이스와의 Connection이 자동으로 종료되지 않는다.
+    따라서, 반드시 Connection 종료를 명시해주어야 **COMMIT**, **ROLLBACK** 과 같은 트랜잭션 연산이 해당 데이터베이스에 반영된다. 
+    즉, Java 저장 함수/프로시저를 호출한 데이터베이스와 실제 연결된 데이터베이스가 다르기 때문에 **별도의 트랜잭션**으로 수행되는 것이다.
```

Comment:
```suggestion
    즉, Java 저장 함수/프로시저를 호출한 데이터베이스와 실제 연결된 데이터베이스가 다르기 때문에 **\별도의 트랜잭션**으로 수행되는 것이다.
```

Reply:
Your intention must be ...

Reply:
I also checked and backslush is not needed.